.github/workflows/build-images-workflow-run.yml

```diff
@@ -336,7 +336,8 @@ jobs:
         if: steps.defaults.outputs.proceed == 'true'
       - name: "Build CI images ${{ matrix.python-version }}:${{ github.event.workflow_run.id }}"
         run: ./scripts/ci/images/ci_prepare_ci_image_on_ci.sh
-        if: matrix.image-type == 'CI' && steps.defaults.outputs.proceed == 'true'
+        # locally built CI image is needed to prepare packages for PROD image build
```

Comment:
What does this do to run time of BuildImage worflows?

Reply:
4 minutes for prod image builds only. For most builds it is 4 minutes for 1 job because we only prepare default image. We need it anyway, because we need consistent environment in which we need to prepare all packages. We could potentially optimise in the future and make a parallel environment for that (we would have to see, fi this is  faster or slower). We can also wait with starting production images  until CI image is ready and pull it - which will be 1 minute overhead,  For now this is the best way if we want to make it before 2.0. There are a few ways we can optimise it. But i want to have ti in place before 2.0 is out. 

That was a deliberate decision.